File: 19. Remove Nth Node From End of List/MySolution.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int val=0, ListNode next=null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */
public class Solution {
    public ListNode RemoveNthFromEnd(ListNode head, int n) {
        var currentNode = head;
        for(int i = 0; i < n; i++) {
            if(currentNode.next == null) return head.next;
            currentNode = currentNode.next;
        }
        var nthPrevNode = head;
        while(currentNode.next != null){
            currentNode = currentNode.next;
            nthPrevNode = nthPrevNode.next;
        }
        nthPrevNode.next = nthPrevNode.next.next;
        return head;
    }
}

File: 19. Remove Nth Node From End of List/MySolution.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @param {number} n
 * @return {ListNode}
 */
var removeNthFromEnd = function(head, n) {
    let currentNode = head;
    for(let i = 0; i<n; i++){
        if(currentNode.next == null) return head.next;
        currentNode = currentNode.next;
    }
    let nthPrevNode = head;
    while(currentNode.next !== null){
        currentNode = currentNode.next;
        nthPrevNode = nthPrevNode.next;
    }
    nthPrevNode.next = nthPrevNode.next.next;
    return head;
};